lib/src/runner/configuration/load.dart

```diff
@@ -51,23 +51,50 @@ Configuration load(String path, {bool global: false}) {
 
   if (document is YamlMap) {
     var loader = new _ConfigurationLoader(document, source, global: global);
-    var config = loader.load();
-    var includePathNode = document.nodes["include"];
-    if (includePathNode != null && includePathNode.value is! String) {
+    return _processIncludeIfAny(loader.load(), document);
+  }
+  throw new SourceSpanFormatException(
+      "The configuration must be a YAML map.", document.span, source);
+}
+
+/// If an `include` node is contained in [node], merges and returns [config].
+Configuration _processIncludeIfAny(Configuration config, YamlMap node) {
+  var source = node.span.sourceUrl.toFilePath(windows: Platform.isWindows);
```

Comment:
I generally use `p.fromUri()` instead of `Uri.toFilePath()`. It's terser, it's more portable, and it's guaranteed to match other path operations' understandings of the current platform. It'll also automatically throw a nice error for non-file URLs. Same below.

Reply:
Wow thanks, did not even know that exists.